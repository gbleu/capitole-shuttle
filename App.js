import React from 'react';
import { StyleSheet, View } from 'react-native';
import { AppLoading } from 'expo';

import Location from './components/Location';
import services from './services';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ecf0f1',
    alignItems: 'center',
    justifyContent: 'center'
  }
});

class App extends React.Component {
  state = {
    appIsReady: false
  }

  componentWillMount() {
    this.bootstrap();
  }

  bootstrap = async () => {
    await services.init();
    console.log('services initialized');
    this.setState({ appIsReady: true });
  };

  render() {
    if (!this.state.appIsReady) {
      return <AppLoading />;
    }

    return (
      <View style={styles.container}>
        <Location />
      </View>
    );
  }
}

export default App;
