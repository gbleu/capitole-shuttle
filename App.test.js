/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

it.skip('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});
