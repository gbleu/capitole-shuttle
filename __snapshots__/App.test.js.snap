// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `<AppLoading />`;

exports[`App component renders as expected 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <App />
</View>
`;
